src/plugin/syncTheme/utils.ts

```diff
@@ -142,3 +180,64 @@ export const checkAndRemoveDeletedSlices = ({
 
   return updatedCurrentVariants;
 };
+
+export enum ErrorType {
+  WRONG_ELEMENT_TYPE = 'wrongElementType',
+  MIXED_SLICE_VALUES = 'mixedSliceValues',
+  DUPLICATED_SLICE_NAME = 'duplicatedSliceName',
+}
+
+export type ErrorMap = Record<ErrorType, string[]>;
+
+/**
+ * Notify the user of each error found stored in the errorMap
+ *
+ * @param errorMap - Object containing the error types and their messages
+ */
+export const notifyErrorMessages = (errorMap: ErrorMap) => {
+  console.log(Object.values(errorMap));
+  Object.values(errorMap)
+    .flat()
+    .forEach((errorMessage) => {
+      figma.notify(`⚠️ Warning: ${errorMessage}`, { timeout: 4000 });
+    });
+};
+
+function createWrongElementMessage({
+  acceptedElementType,
+  currentType,
+  sliceName,
+}: {
+  acceptedElementType: string;
+  currentType: string;
+  sliceName: string;
+}) {
+  return `The ${sliceName} slice accepts only ${acceptedElementType} elements, you're also using ${currentType}`;
+}
+
+function createDuplicatedSliceNameMessage({ variantName, sliceName }: { variantName: string; sliceName: string }) {
+  return `The ${variantName} variant on ${sliceName} already exist, please use a unique name`;
+}
+
+function createMixedSliceValuesMessage({ variantName, sliceName }: { variantName: string; sliceName: string }) {
+  return `You're using mixed ${sliceName} in ${variantName}. Please use a single value`;
+}
+
+const warningMessageGenerators = {
+  [ErrorType.WRONG_ELEMENT_TYPE]: createWrongElementMessage,
+  [ErrorType.DUPLICATED_SLICE_NAME]: createDuplicatedSliceNameMessage,
+  [ErrorType.MIXED_SLICE_VALUES]: createMixedSliceValuesMessage,
+};
+
+export function generateAndRecordErrorMessage<T extends ErrorType>({
+  errorType,
+  errorMap,
+  values,
+}: {
+  errorType: T;
+  errorMap: ErrorMap;
+  values: Parameters<typeof warningMessageGenerators[T]>[0];
+}) {
+  const selectedMessage = warningMessageGenerators[errorType](values as any);
```

Comment:
**question:** any chance to have a better type on `values`? I would avoid `as any` if possible